### DISCUSSION TITLE
HEADER Properties

### DISCUSSION THREAD

**Discussion by RyanDaley** [GeometryGym/GeometryGymIFC#79] [Q&A] [answered]
I'm wondering if its possible to use GeometryGym to access the information in the Header such as FILE_DESCRIPTION, FILE_NAME, FILE_SCHEMA.  For now, I would just like to read this information.

I want to write something like:
`db.OfType<GeometryGym.Ifc.FileSchema>().FirstOrDefault();`  

I just can't find out what these types might be called.  So I can't figure out how to retrieve them.

![image](https://user-images.githubusercontent.com/65096230/199261100-92ac6465-8a6e-4660-a2e0-9d5774a960de.png)

**Comment by RyanDaley** [comment]
@jmirtsch  Can GeometryGym be used for this?

**Comment by jmirtsch** [comment, marked as answer]
Hi Ryan,

I added a new attribute to the database with contents read from step file header.  
https://github.com/GeometryGym/GeometryGymIFC/blob/09a0f173b59312bb9c548da8c61eb7677eaf0dde/Core/STEP/DatabaseSTEP.cs#L40

I'm not sure if your preference might be to only read the header (and ignore all the data), this would need a new method if desired.
Given the size of many IFC data sets, parsing the entire file might not be required.

Hope it helps,

Jon
